src/python/pants/rules/core/fmt_test.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import List
+
+from pants.engine.fs import EMPTY_DIRECTORY_DIGEST, Digest, FilesContent
+from pants.engine.legacy.graph import HydratedTarget, TransitiveHydratedTargets
+from pants.engine.legacy.structs import PythonTargetAdaptor
+from pants.engine.rules import UnionMembership
+from pants.option.option_value_container import OptionValueContainer, RankedValue
+from pants.rules.core.fmt import Fmt, FmtResult, TargetWithSources, fmt
+from pants.testutil.engine.util import MockConsole, MockGet, run_rule
+
+
+def test_transitive_option() -> None:
+  dependency = HydratedTarget(
+    address="src/dep",
+    adaptor=PythonTargetAdaptor(sources=(), name="dep"),
+    dependencies=()
+  )
+  root = HydratedTarget(
+    address="src/root",
+    adaptor=PythonTargetAdaptor(sources=(), name="root"),
+    dependencies=(dependency,)
+  )
+  targets = TransitiveHydratedTargets(closure=(dependency, root), roots=(root,))
+
+  def assert_targets_formatted(
+    *, transitive: bool, expected_targets: List[HydratedTarget]
+  ) -> None:
+    console = MockConsole(use_colors=False)
+
+    # TODO: make it more ergonomic in tests to set an @console_rules's Options when using
+    #  engine.util.run_rule().
+    options = OptionValueContainer()
+    options.transitive = RankedValue(RankedValue.HARDCODED, transitive)
```

Comment:
This is painful to mock `Goal.Options` currently, but doesn't need to be! Planning on having a followup to add a util for this with a signature like this: `MockGoalOptions(goal: Goal, values: Dict[str, Any]) -> Goal.Options`